### ISSUE TITLE
users hit stop work instead of submit work after completing hackathon prizes which messes up data collection

### ISSUE THREAD

**Issue by frankchen07** [gitcoinco/web#8686]
### Circumstances

When developers start work on a hackathon issue, instead of correctly hitting submit work, they mistakenly hit "stop" work because they assume that's what one is supposed to do after completing the issue. 

This messes up the data collected in our DB, and makes it difficult for us to correctly associate work started interest, work submitted, and submission links for our sponsors.

### Description

The correct way to fix this would be to re-think the bounties architecture but a quick fix is to modify our user's behavior with better language positioning. 

Suggested:

Start Work
Submit Work
Quit Bounty

**Comment by willsputra** [comment]
we could:
• change the wording to Quit Work. Quit Bounty might be confusing if we're moving hackathons away from the bounties architecture.
• change the Quit Work button to red to indicate a negative action.
• adding a confirmation popup (we could also ask users why they quit work here, if we want)
![Slice 1](https://user-images.githubusercontent.com/19514207/112495586-fc3b9d00-8dbe-11eb-8e5b-1d9247cc9696.png)

**Comment by frankchen07** [comment]
Sounds like a reasonable stopgap solution. Anything else assets-wise required to bounty this?

**Comment by PixelantDesign** [comment]
Conceptually Quit work makes sense. 

**Quit Work Modal**

Messaging

Is it true that quitting work will exclude the user from all prizes? Some sponsors pay for the ideas and don't care as much about implementation. I wonder if there is a way to frame that differently.

For the message:
If you plan on submitting work, do not click quit work -- could we make this look more like a warning? Perhaps different color treatment/alert like. Just an idea.

Since **Quit Work** a descructive action, let's use the new Gitcoin red color to confirm the action. 

For the third statement "If you plan on quitting work, do not quit work." We should try to be explicit on the alternative action to Quit Work. Perhaps a secondary action button with the word Cancel.

A few other things I'm thinking about... is it possible to center the modal? What does the responsive view look like?

**Other Scenarios**

Where else might we need to consider updating this info? Hackathon onboarding?

Should there be logic that checks whether the person is working by themselves or part of a team? Bounty funder/teammates should likely be notified if a person has quit work. What happens on the project card in the hackathon view? 

What do we show on the activity feed on bounty detail?

We will also need to update the email that the funder receives that a worker has "Quit Work".

**Comment by willsputra** [comment]
updated the banners
![Frame 3](https://user-images.githubusercontent.com/19514207/112986060-9ae74580-9193-11eb-9463-f6682fe827dd.jpeg)

**Comment by gdixon** [comment]
Should this message: 

"If you plan on quitting work, do not quit work." 

Read:

"If you plan on **submitting** work, do not _quit work_."

**Comment by willsputra** [comment]
> "If you plan on submitting work, do not quit work."

yes! sorry my bad

**Comment by PixelantDesign** [comment]
Agree with @gdixon feedback on the copy. 

@willsputra could we confirm the button placement within the modal with the design system? Also, is it possible to center the modal? What happens after the user clicks Quit Work? We would display the success banner?

**Comment by frankchen07** [comment]
The main problem we're aiming to solve with this one is stopping the disruption of folks who hit "stop/quit" work and then submit anyway. It messes up the data extracts that Connor and I provide because it removes interest, which is one of the necessary artifacts we join on.

This will be important to understand when we help to automate https://github.com/gitcoinco/web/issues/8687 for the sponsor.

> Should there be logic that checks whether the person is working by themselves or part of a team? 

Ideally, yes, but given the scope of our problem I don't think a logic check of this caliber is necessary.

> Bounty funder/teammates should likely be notified if a person has quit work. 

I think just changing the language in the activity feed would be good, as well as the email that notifies a funder if someone has stopped work. That might just necessitate copy changes.

> What happens on the project card in the hackathon view?

I believe the project will remain because that's of a different data model from interest. If I have it right and quitting work removes interest, then the way we construct the data output should also naturally eliminate folks who quit work, so the data extract in https://github.com/gitcoinco/web/issues/8687 would be correct, but a "dead" project will remain. It's not optimal but I'm not too worried about it.

**Comment by connoroday** [comment]
I agree that the proposed solution would be a big improvement. The change of language and confirmation pop-up would definitely help alleviate most issues, but maybe not all.

A full fix would be if someone stops then starts again it doesn’t mess up the data collection, and additionally, once someone makes a submission, “stop work” is no longer an option at all.

Pasting my comment from the word doc here, but the thought process that leads to issues usually are:

“I completed my project and was finished working so I naturally clicked stop work”

“After I submitted my project I clicked ‘stop work’ because I was no longer working”

“I clicked it by accident/didn’t understand”

I think changing the language to “Quit bounty” or “Quit working” would alleviate the issue a bit, and a pop up when someone clicks stop work that says “Are you sure you want to give up working on this bounty for good? (Don’t click if you may plan to submit)” etc

Which is pretty much what is being done here, so I'm on board!

**Comment by willsputra** [comment]
> @willsputra could we confirm the button placement within the modal with the design system? Also, is it possible to center the modal? What happens after the user clicks Quit Work? We would display the success banner?

@PixelantDesign updated with the design system modal. yes, once a user clicks Quit Work, we could display success banner to indicate that they have successfully quit work.

![Slice 1](https://user-images.githubusercontent.com/19514207/113291343-7ae78b80-9325-11eb-9089-54e21e72f40d.png)

> I think just changing the language in the activity feed would be good, as well as the email that notifies a funder if someone has stopped work. That might just necessitate copy changes.

agreed! we can change all the copy from Stop Work to Quit Work.

**Comment by frankchen07** [comment]
I want to bring @octavioamu into this conversation before I bounty this out. Reviewing the scope:

- change the language from stop work to quit work
- updated quit work modal upon clicking
- changing the language in the activity feed from "user stopped work" to "user quit work"
- changing the email language to the funder that a user has quit work
- not included in the scope is anything dealing with the project model, we can leave that alone

perhaps we can grab a 15-min convo sometime @octavioamu
